src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -301,11 +301,7 @@ private function parseXmlAttributes(\DOMNode $node)
         $data = array();
 
         foreach ($node->attributes as $attr) {
-            if (ctype_digit($attr->nodeValue)) {
-                $data['@'.$attr->nodeName] = (int) $attr->nodeValue;
-            } else {
-                $data['@'.$attr->nodeName] = $attr->nodeValue;
-            }
+            $data['@'.$attr->nodeName] = $attr->nodeValue;
```

Comment:
@jvasseur yes, it retains BC, but will work only for positive integers, as I wrote above.
I vote to break BC, but always returned string.